features/filesystem/littlefs/LittleFileSystem.cpp

```diff
@@ -164,36 +165,40 @@ int LittleFileSystem::mount(BlockDevice *bd)
     }
 
     err = lfs_mount(&_lfs, &_config);
-    LFS_INFO("mount -> %d", lfs_toerror(err));
+    if (err) {
+        _bd = NULL;
+        LFS_INFO("mount -> %d", lfs_toerror(err));
+        _mutex.unlock();
+        return lfs_toerror(err);
+    }
+
     _mutex.unlock();
-    return lfs_toerror(err);
+    LFS_INFO("mount -> %d", 0);
+    return 0;
 }
 
 int LittleFileSystem::unmount()
 {
     _mutex.lock();
     LFS_INFO("unmount(%s)", "");
+    int res = 0;
     if (_bd) {
         int err = lfs_unmount(&_lfs);
-        if (err) {
-            LFS_INFO("unmount -> %d", lfs_toerror(err));
-            _mutex.unlock();
-            return lfs_toerror(err);
+        if (err && !res) {
```

Comment:
Just a nit: !res check is not really needed here

Reply:
Yeah, I just kept it for consistency. That way if any code is added between the declaration of res and this line, it won't break anything.

The compiler will optimize it out anyways.

Reply:
> The compiler will optimize it out anyways.

Famous last words